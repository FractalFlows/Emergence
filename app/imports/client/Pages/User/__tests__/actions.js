import {
  loginInit,
  loginCheck,
  loginError,
  loginSuccess,
  loginClean,
  refreshToken,
  userFetchData,
} from '../actions'

describe('User actions', () => {
  it('loginInit should create a proper action', () => {
    const action = loginInit()

    expect(action).toMatchSnapshot()
  })

  it('loginSuccess should create a proper action', () => {
    const action = loginSuccess({
      token: 'astrocoders_rocks_and_this_is_not_a_secret',
    })

    expect(action).toMatchSnapshot()
  })

  it('loginCheck should create a proper action', () => {
    const action = loginCheck({
      routeName: 'some_neat_route',
    })

    expect(action).toMatchSnapshot()
  })

  it('loginClean should create a proper action', () => {
    const action = loginClean()

    expect(action).toMatchSnapshot()
  })

  it('refreshToken should create a proper action', () => {
    const action = refreshToken({
      token: 'astrocoders_rocks_and_this_is_not_a_secret',
    })

    expect(action).toMatchSnapshot()
  })

  it('userFetchData should create a proper action', () => {
    const action = userFetchData({
      data: {
        id: 'x',
        name: 'Edward Elric',
        level: 'alchemist',
      }
    })

    expect(action).toMatchSnapshot()
  })
})
