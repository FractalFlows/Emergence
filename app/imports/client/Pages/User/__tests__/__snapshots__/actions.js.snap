exports[`User actions loginCheck should create a proper action 1`] = `
Object {
  "payload": Object {
    "routeName": "some_neat_route",
  },
  "type": "user/USER_LOGIN_CHECK",
}
`;

exports[`User actions loginClean should create a proper action 1`] = `
Object {
  "type": "user/USER_LOGIN_CLEAN",
}
`;

exports[`User actions loginInit should create a proper action 1`] = `
Object {
  "type": "user/USER_LOGIN_INIT",
}
`;

exports[`User actions loginSuccess should create a proper action 1`] = `
Object {
  "payload": Object {
    "token": "astrocoders_rocks_and_this_is_not_a_secret",
  },
  "type": "user/USER_LOGIN_SUCCESS",
}
`;

exports[`User actions refreshToken should create a proper action 1`] = `
Object {
  "payload": Object {
    "token": "astrocoders_rocks_and_this_is_not_a_secret",
  },
  "type": "user/USER_REFRESH_TOKEN",
}
`;

exports[`User actions userFetchData should create a proper action 1`] = `
Object {
  "payload": Object {
    "me": Object {
      "id": "x",
      "level": "alchemist",
      "name": "Edward Elric",
    },
  },
  "type": "user/USER_FETCH_DATA",
}
`;
